packages/expect/src/__tests__/matchers-toStrictEqual.property.test.ts

```diff
@@ -46,4 +56,18 @@ describe('toStrictEqual', () => {
       assertSettings,
     );
   });
+
+  it('should be equivalent to Node deepStrictEqual', () => {
+    fc.assert(
+      fc.property(
+        fc.anything(anythingSettings),
+        fc.anything(anythingSettings),
```

Comment:
As we want to fully mimic the behaviour of `assert.deepStrictEqual` whatever the inputs, we might want to increase the scope of values that can be generated by the generator. To do that you can replace `anythingSettings` by `{ ...anythingSettings, withMap: true, withSet: true, withObjectString: true }`.

As stated in https://github.com/facebook/jest/blob/59e40efb5be038079fde102017da0db719728580/packages/expect/src/__tests__/__arbitraries__/sharedSettings.ts, `withSet` and `withMap` are currently disabled because of an opened issue.

If you update the scope of possible inputs - as suggested above - you might force a re-run of fast-check with more tests just to make sure nothing will break soon in the CI. To do that you can add an extra setting temporary to `fc.assert(/*your fc.property*/, {numRuns: 1000000})`

Reply:
Thanks for the review! I imagine the `Set` / `Map` bugs will also fail this if Node behaves correctly on those, so we have keep the restricted scope here as well? I'll try it out

Reply:
Yeah it finds a failure with `Boolean`s in `Set`s